src/web/app/views/beambox/Right-Panels/ObjectPanelItem.tsx

```diff
@@ -0,0 +1,210 @@
+import classNames from 'classnames';
+import Icon from '@ant-design/icons';
+import React, { useContext } from 'react';
+import { Button, Divider, Mask, Popover } from 'antd-mobile';
+
+import ObjectPanelIcons from 'app/icons/object-panel/ObjectPanelIcons';
+import storage from 'implementations/storage';
+import units from 'helpers/units';
+import { getSVGAsync } from 'helpers/svg-editor-helper';
+import { ObjectPanelContext } from 'app/views/beambox/Right-Panels/contexts/ObjectPanelContext';
+
+import styles from './ObjectPanelItem.module.scss';
+
+let svgEditor;
+getSVGAsync((globalSVG) => {
+  svgEditor = globalSVG.Editor;
+});
+
+interface Props {
+  id: string;
+  content: JSX.Element;
+  label?: string | JSX.Element;
+  onClick?: () => void;
+  disabled?: boolean;
+}
+const ObjectPanelItem = ({ id, content, label, onClick, disabled }: Props): JSX.Element => {
+  const context = useContext(ObjectPanelContext);
+  const { activeKey, updateActiveKey } = context;
+  if (disabled) {
+    return null;
+  }
+  return (
+    <div
+      className={classNames(styles['object-panel-item'], {
+        [styles.active]: activeKey === id,
+      })}
+      onClick={() => {
+        updateActiveKey(id);
+        onClick?.();
+      }}
+    >
+      <div className={styles.main}>{content}</div>
+      {label && <div className={styles.label}>{label}</div>}
+    </div>
+  );
+};
+
+interface NumberItemProps {
+  id: string;
+  value: number;
+  updateValue?: (val: number) => void;
+  label?: string | JSX.Element;
+}
+const ObjectPanelNumber = ({ id, label, value, updateValue }: NumberItemProps): JSX.Element => {
+  const context = useContext(ObjectPanelContext);
+  const { activeKey, updateActiveKey } = context;
+  const isActive = activeKey === id;
+  const useInch = id !== 'rotate' && storage.get('default-units') === 'inches';
+  const unit = useInch ? 'inch' : 'mm';
+  const precision = useInch ? 4 : 2;
+  const valueInUnit = (+units.convertUnit(value, unit, 'mm').toFixed(precision)).toString();
```

Comment:
I thought `toFixed` would return string, and then unary + cast it into number and finally `toString` cast it back into string again.
Can we just use `units.convertUnit(value, unit, 'mm').toFixed(precision)`?